### PR TITLE
[IMP] mail: introduce mail services module

### PR DESCRIPTION
**Pull request by phenix-factory**

### CODE REVIEW

File: addons/mail/tooling/types/services.d.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        "mail.activity": ReturnType<typeof activityService.start>;
+        "mail.core.web": ReturnType<typeof mailCoreWeb.start>;
+        "discuss.channel.member": ReturnType<typeof channelMemberService.start>;
+        "discuss.rtc": ReturnType<typeof rtcService.start>;
+        "discuss.core.common": ReturnType<typeof discussCoreCommon.start>;
+        "discuss.core.public": ReturnType<typeof discussCorePublic.start>;
+        "discuss.core.web": ReturnType<typeof discussCoreWeb.start>;
+        "discuss.gifPicker": ReturnType<typeof gifPickerService.start>;
+        "discuss.message.pin": ReturnType<typeof messagePinService.start>;
+        "discuss.typing": ReturnType<typeof discussTypingService.start>;
```
Comment (seb-odoo) [inline addons/mail/tooling/types/services.d.ts:52]: (minor) I would sort them alphabetically by service name

**Review by seb-odoo — APPROVED** [review]
I didn't test the autocomplete but the diff looks ok.

I'm afraid of conflict with voice PR (which has priority), so I'll r+ this later.

**Comment by seb-odoo** [comment]
Conflicts can be resolved now

**Comment by phenix-factory** [comment]
> Conflicts can be resolved now

Conflict resolved :)

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**